### ISSUE TITLE
Clarification about parsing-logic

### ISSUE THREAD

**Issue by mabruzzo** [enzo-project/enzo-e#392]
@jobordner -- I've recently been compiling with Apple's Clang compiler and it raised a noteworthy warning about a codesnippet in the parsing code.

Specifically, it points out that: [this line](https://github.com/enzo-project/enzo-e/blob/54f4f00e0a33ea96fbff498c31a1d83ca2c52620/src/Cello/parse.y#L849) always evaluates to true.

In more detail:
- the line reads

   ```c
     if (p->group != NULL) {
   ```

- For context `p` is a pointer to [struct param_struct](https://github.com/enzo-project/enzo-e/blob/54f4f00e0a33ea96fbff498c31a1d83ca2c52620/src/Cello/parse.h#L61). In that struct the `group` member is defined as `char * group[MAX_GROUP_DEPTH];`.
- In other words `p->member` can't be true

It was not immediately obvious to me what's going on here. Should we just remove the `else` clause here? Or is this a typo (for example instead maybe it should read `p->group[0] != NULL`)?
